IBPSA/Fluid/FixedResistances/CheckValve.mo

```diff
@@ -0,0 +1,143 @@
+within IBPSA.Fluid.FixedResistances;
+model CheckValve "Hydraulic one way valve"
+
+  extends IBPSA.Fluid.BaseClasses.PartialResistance(
+    dp(nominal=6000),
+    final dp_nominal=dpValve_nominal + dpFixed_nominal,
+    final m_flow_turbulent=deltaM*abs(m_flow_nominal),
+    final from_dp=false,
+    final linearized=false);
+
+  extends IBPSA.Fluid.Actuators.BaseClasses.ValveParameters(
+    rhoStd=Medium.density_pTX(101325, 273.15 + 4, Medium.X_default));
+
+  parameter Modelica.SIunits.PressureDifference dpFixed_nominal(
+    displayUnit="Pa", min=0) = 0
+    "Pressure drop of pipe and other resistances that are in series"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Real l(min=1e-10, max=1)=0.001 "Valve leakage, l=Kv(y=0)/Kv(y=1)";
+
+  parameter Real kFixed(unit="", min=0)=
+    if dpFixed_nominal > Modelica.Constants.eps then
+      m_flow_nominal/sqrt(dpFixed_nominal)
+    else 0
+    "Flow coefficient of fixed resistance that may be in series with valve, 
+    k=m_flow/sqrt(dp), with unit=(kg.m)^(1/2).";
+
+  Real k(unit="", min=Modelica.Constants.small)
+    "Flow coefficient of valve and pipe in series in allowed/forward direction, 
+    k=m_flow/sqrt(dp), with unit=(kg.m)^(1/2).";
+
+protected
+  Real a     "Pressure scaled variable";
+  Real cv    "Smoothed heaviside checkvalve characteristic";
+  Real kstar "Smoothed restriction characteristic";
+
+initial equation
+  assert(dpFixed_nominal > -Modelica.Constants.eps,
+    "In " + getInstanceName() + ": Require dpFixed_nominal >= 0. 
+    Received dpFixed_nominal = " + String(dpFixed_nominal) + " Pa.");
+
+  assert(l > -Modelica.Constants.eps,
+    "In " + getInstanceName() + ": Require l >= 0. Received l = " + String(l));
+
+equation
+
+  a = dp/dpValve_nominal;
+  cv = max(0,min(1,a^3*(10+a*(-15+6*a)))); // update this to call
+  //IBPSA.Utilities.Math.Functions.smoothHeaviside when issue 1202 is merged
+  kstar = Kv_SI*(cv + l);
+
+  // add series restriction when applicable
+  if (dpFixed_nominal > Modelica.Constants.eps) then
+    k = sqrt(1/(1/kFixed^2 + 1/kstar^2));
+  else
+    k = kstar;
+  end if;
+
+  if homotopyInitialization then m_flow = homotopy(actual=
+      IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp(
+        dp = dp, k = k, m_flow_turbulent = m_flow_turbulent),
+      simplified = if dp > 0 then k*dp else k*l*dp);

```

Comment:
This should be `then m_flow_nominal*dp/dp_nominal else ...` to be consistent with`IBPSA.Fluid.Actuators.BaseClasses.PartialTwoWayValveKv`.